### PR TITLE
277 radius not acc pw

### LINKED ISSUE #277: Check radius pw is not main acct pw

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes #277 this prevents the radius password, or variations of it, being used as the main account or unix credential.

- [ x ] cargo fmt has been run
- [ - ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ - ] book chapter included (if relevant)
- [ - ] design document included (if relevant)

### CODE REVIEW

File: kanidmd/src/lib/idm/server.rs [hunk 1]
```diff
@@ -955,12 +955,16 @@ impl<'a> IdmServerProxyWriteTransaction<'a> {
         // I don't think so - because we should only be showing how STRONG the pw is ...
```
Comment (yaleman) [inline kanidmd/src/lib/idm/server.rs:955]: *I* think so... enumerating passwords is possible if that's a thing :)

**Review by yaleman — CHANGES_REQUESTED** [review]
Yeah, if the system's going to compare against an existing password and provide feedback to the user that it's invalid (even if it doesn't say that) then ... it's possible to infer a current password, without rate limits. Maybe put *that* check after the "can the user make the change" permission check?

**Comment by Firstyear** [comment]
Done, it's possible now to do a partial mod application which checks everything "up until" applying to the DB. This means access, schema, plugins are run, then we check the quality, then we finish the application. This avoids the attack method you have noted here.

**Review by yaleman — APPROVED** [review]
Sweet, all checks out from my reading of it. Also - comments explaining the flow are much appreciated as someone Outside Williams Head trying to understand it. :)

**Pull request merged**